### PR TITLE
feat: svelte support

### PR DESCRIPTION
**Pull request by thenbe**
### Description

This PR adds svelte support (as per https://github.com/antfu/eslint-config/issues/282#issuecomment-1817883159)

It includes the following rules:

1. Stylistic rules for users not using prettier
1. The recommended rule-set from [eslint-plugin-svelte](https://sveltejs.github.io/eslint-plugin-svelte/rules/)
1. Rules that users always need to amend for svelte anyways, such as:

   - setting `varsIgnorePattern` for svelte specific stuff such as `$$Props`
   - turning off `import/no-mutable-exports` because svelte uses `export let myprop` to declare props
   - etc

1. Some additional rules from eslint-plugin-svelte that are likely to indicate an issue (I tried to avoid the more opinionated ones)

---

### Additional context

A few points:

- The [`Rules`](https://github.com/thenbe/eslint-config/blob/svelte-config/src/types.ts?plain=1#L35) type does not include svelte rules because they are not a part of https://github.com/antfu/eslint-define-config.
- For the dependency setup: Users are only prompted to install `eslint-plugin-svelte`. They don't need to install `svelte-eslint-parser` directly, since it's already a dependency of `eslint-plugin-svelte`.
  - NOTE: I had to include `svelte-eslint-parser` as a peer dependency. Without it, this project will fail to build (it errors about not finding `svelte`). And if we install `svelte` and remove `svelte-eslint-parser`, we can build but we get an error when _consuming_ this package `Error: Dynamic require of "fs" is not supported`. This current setup builds successfully and can be consumed successfully.

### CODE REVIEW

File: src/configs/svelte.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        'no-undef': 'off', // incompatible with most recent (attribute-form) generic types RFC
+        'no-unused-vars': ['error', {
+          args: 'none',
+          caughtErrors: 'none',
+          ignoreRestSiblings: true,
+          vars: 'all',
+          varsIgnorePattern: '^\\$\\$Props$',
+        }],
+
+        ...pluginSvelte.configs.recommended.rules as any,
```
Comment (antfu) [inline src/configs/svelte.ts:61]: I wonder how many rules are in it? Should we just expand it inline so have a clear image what to include and compose (for example, would it include some rules in the stylistic category?)

**Review by antfu — COMMENTED** [review]
Thanks! In general I am happy to have it, if you are willing to help maintaining it (like I might ping you when ppl raise Svelte related issues), is that ok for you? Thanks

File: src/configs/svelte.ts (hunk 1 above)
Comment (thenbe) [reply src/configs/svelte.ts:61]: There are [14](https://github.com/sveltejs/eslint-plugin-svelte/blob/d22f6e957f00c2046f72e46d825727ad5ff2f658/src/configs/recommended.ts?plain=1#L11-L24) rules marked as recommended today, none of which are categorized as [stylistic](https://sveltejs.github.io/eslint-plugin-svelte/rules/#stylistic-issues).

I don't have an opinion on this issue, we can be more explicit by expanding the rules.

**Comment by thenbe** [comment]
Sure thing!

File: src/configs/svelte.ts (hunk 1 above)
Comment (thenbe) [reply src/configs/svelte.ts]: expanded: https://github.com/antfu/eslint-config/pull/364/commits/9e31d9d48ebfc7bc463881f826e4e8f27155a234

**Pull request merged**